runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/PTransformMatchers.java

```diff
@@ -212,7 +216,14 @@ public boolean matches(AppliedPTransform<?, ?, ?> application) {
         }
         CreatePCollectionView<?, ?> createView =
             (CreatePCollectionView<?, ?>) application.getTransform();
-        ViewFn<Iterable<WindowedValue<?>>, ?> viewFn = createView.getView().getViewFn();
+        PCollectionView<?> view = createView.getView();
+        checkState(
+            view instanceof SimplePCollectionView,
+            "Unknown %s type: %s",
```

Comment:
Make it actionable by saying what is expected (`"only SimplePCollectionView is supported"`)